mkosi/__init__.py

```diff
@@ -2621,161 +2552,35 @@ def install_arch(args: CommandLineArguments, root: Path, do_run_build_script: bo
         hooks_dir.joinpath("60-depmod.hook").symlink_to("/dev/null")
 
         hooks_dir.joinpath("90-mkosi-kernel-add.hook").write_text(
-            dedent(
-                """\
-                [Trigger]
-                Operation = Install
-                Operation = Upgrade
-                Type = Path
-                Target = usr/lib/modules/*/vmlinuz
-                Target = usr/lib/kernel/install.d/*
-                Target = boot/*-ucode.img
-
-                [Trigger]
-                Operation = Install
-                Operation = Upgrade
-                Type = Package
-                Target = systemd
-
-                [Action]
-                Description = Adding kernel and initramfs images to /boot...
-                When = PostTransaction
-                Exec = /etc/pacman.d/scripts/mkosi-kernel-add
-                NeedsTargets
-                """
-            )
+            importlib.resources.read_text("mkosi.resources.arch", "60_kernel_remove.hook")
         )
```

Comment:
I think it'd make sense to add a helper for this:
```py
def write_resource(where: Path, resource: str, key: str, bool: executabe = False) -> None:
    text = importlib.resources.read_text(resource, key)
    where.write_text(text)
    if executable: …

write_resource(hooks_dir / '90-mkosi-kernel-add.hook', 'mkosi.resources.arch', '60_kernel_remove.hook')
```

```

Reply:
Will change it accordingly.

Reply:
Added a `mode` switch as well.